vpr/src/base/vpr_context.h

```diff
@@ -416,6 +417,8 @@ struct FloorplanningContext : public Context {
      * The constraints on each cluster are computed during the clustering process and can change.
      */
     vtr::vector<ClusterBlockId, PartitionRegion> cluster_constraints;
+
+    AttractionInfo attraction_groups;
```

Comment:
Might be better in the clustering context.

Reply:
Clear the attraction groups after clustering and comment that you are doing so. (Would lead to wasted memory if no one else needs it after clustering).

Reply:
Actually would be better if you create it early (during prepacking) and then pass it around by reference to the routines that need it during clustering. Can find some different structures that are all passed around during clustering and group them together so that you don't have to add it individually to too many parameter lists. Can call it during cl_optimize or something along those lines.

Can create it in a header file (probably cluster.h). Can just make it a struct with no functions (or maybe have a constructor and destructor if they are all created and destroyed and the same time).